### ISSUE TITLE
Bake Ambient Occlusion ?

### ISSUE THREAD

**Issue by Inspirateur** [Inspirateur/binary-greedy-meshing#2]
The [v.1.0.0 branch](https://github.com/cgerikj/binary-greedy-meshing/tree/v1.0.0) of the C library I've ported bakes Ambient Occlusion in the vertices. 

However, the new version only returns quads, because that's more efficient with instancing & batching and you can deduce the vertices from the quad data anyway. We could store an AO value for each quad, it should only take 4 bits (1/0 for each corner).

**Comment by Inspirateur** [comment]
Actually unsure if it is a good idea because it does cost a lot in meshing time and we might have it for cheaper with screen space techniques

**Comment by MountainLabz** [comment]
Ehh, speed over AO please :)

**Comment by Aceeri** [comment]
I think this might still be worthwhile as an option, SSAO only gives a small amount, which does make it still a bit hard to see corners of voxels. Going to look into implementing this in the future after I get some gameplay going for my project.

SSAO doesn't really provide much for the scenario of where you look straight down above voxels, which makes it harder to read the elevation change.

E.g. from spacefarer's vertex AO rather than SSAO:
![Image](https://github.com/user-attachments/assets/e08ac75f-69e2-4d9e-a030-1b9086b16a33)

**Comment by Inspirateur** [comment]
Yeah I still don't know what's the best approach, there might be a 3rd option still that we don't know of. I'll ask Gore for his wisdom.
